learn/cookbooks/laravel-multitenancy.mdx

```diff
@@ -0,0 +1,365 @@
+
+You can choose when you want to generate the token. Here, we will generate the token when the organization is retrieved from the database. If the organization has no token, we will generate one and store it in the `meilisearch_token` attribute.
+
+Let's update the code in our `app/Models/Organization.php` code:
+
+```php
+<?php
+
+namespace App\Models;
+
+use DateTime;
+use Laravel\Scout\EngineManager;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use Illuminate\Support\Facades\Log;
+
+class Organization extends Model
+{
+
+    public function contacts(): HasMany
+    {
+        return $this->hasMany(Contact::class);
+    }
+
+    protected static function booted()
+    {
+        static::retrieved(function (Organization $organization) {
+            // We can add some logic to skip generating tokens in certain environments
+            if (env('SCOUT_DRIVER') === 'array' && env('APP_ENV') === 'testing') {
+                $organization->meilisearch_token = 'fake-tenant-token';
+                return;
+            }
+
+            // Early return if the organization already has a token
+            if ($organization->meilisearch_token) {
+                Log::debug('Organization ' . $organization->id . ': already has a token');
+                return;
+            }
+            Log::debug('Generating tenant token for organization ID: ' . $organization->id);
+
+            // The object belows is used to generate a tenant token that:
+            // • applies to all indexes
+            // • filters only documents where `organization_id` is equal to this org ID
+            $searchRules = (object) [
+                '*' => (object) [
+                    'filter' => 'organization_id = ' . $organization->id,
+                ]
+            ];
+
+            // Replace with your own Search API key and API key UID
+            $meiliApiKey = env('MEILISEARCH_SEARCH_KEY');
+            $meiliApiKeyUid = env('MEILISEARCH_SEARCH_KEY_UID');
+
+            // Generate the token
+            $token = self::generateMeiliTenantToken($meiliApiKeyUid, $searchRules, $meiliApiKey);
+
+            // Save the token in the database
+            $organization->meilisearch_token = $token;
+            $organization->save();
+        });
+    }
+
+    protected static function generateMeiliTenantToken($meiliApiKeyUid, $searchRules, $meiliApiKey)
+    {
+        $meilisearch = resolve(EngineManager::class)->engine();
+
+        return $meilisearch->generateTenantToken(
+            $meiliApiKeyUid,
+            $searchRules,
+            [
+                'apiKey' => $meiliApiKey,
+                'expiresAt' => new DateTime('2030-12-31'),
+            ]
+        );
+    }
+}
+```
+
+Now that our `Organization` model is generating tenant tokens, we need to provide the front-end with these tokens so that it can access Meilisearch securely.
```

Comment:
```suggestion
Now the `Organization` model is generating tenant tokens, you need to provide the front-end with these tokens so that it can access Meilisearch securely.
```